### ISSUE TITLE
Failed Keras Import using a Multi Inputs/Outputs model from the Keras Functional API. 

### ISSUE THREAD

**Issue by claeyzre** [deeplearning4j/deeplearning4j#4751]
#### Issue Description
I built a Keras Model using the Keras Functional API. This is a multi Input/Output model inspired from this [publication](https://arxiv.org/pdf/1704.05146.pdf). Basically 8 different Inputs which consist in 4 categorical inputs and 4 textual inputs (indexed sequences). 
The textual inputs which consists of sequences of indices, are then fed into 4 different trainable Embeddings layers. the outputs of those layers are fed into convolutions and then flattened. 
When it's done the results of the 4 convolutions and the 4 categorical variables are concatenated and fed into a Dense layer with 60 hidden dims.  The results go through two different outputs (city and country) which consists in two Dense layer with Softmax activation used for prediction.

The keras model train and predict fine in Keras(even after exporting the h5 and json and reloading it from those files), but I got this stacktrace when importing the network in DL4J:
`Exception in thread "main" java.lang.IllegalStateException: Mis matched lengths: [56160] != [163680] - Array 1 shape: [936, 60], array 2 shape: [2728, 60]
	at org.nd4j.linalg.util.LinAlgExceptions.assertSameLength(LinAlgExceptions.java:41)
	at org.nd4j.linalg.api.ops.BaseTransformOp.<init>(BaseTransformOp.java:174)
	at org.nd4j.linalg.api.ops.impl.transforms.Set.<init>(Set.java:41)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.assign(BaseNDArray.java:1270)
	at org.deeplearning4j.nn.layers.BaseLayer.setParam(BaseLayer.java:204)
	at org.deeplearning4j.nn.modelimport.keras.KerasLayer.copyWeightsToLayer(KerasLayer.java:295)
	at org.deeplearning4j.nn.modelimport.keras.utils.KerasModelUtils.copyWeightsToModel(KerasModelUtils.java:76)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.getComputationGraph(KerasModel.java:376)
	at org.deeplearning4j.nn.modelimport.keras.KerasModel.getComputationGraph(KerasModel.java:362)
	at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasModelAndWeights(KerasModelImport.java:180)
`

when calling
`ComputationGraph network = KerasModelImport
                .importKerasModelAndWeights("model-v3.json",
                        "model-v3.h5", false);`

I set enforceTraining to false because I wish to do Inference only.

I know that this is a problem of dimension but I don't know where DL4J got the value `936`, as the results of the concatenation leads to a shape of `2728`.

I Used Input, Embedding, Conv1D MaxPooling1D, Concatenate and Dense as Layers. I also built a version without the Embedding, feeding them directly in Input but DL4J throws the same exception
I used the 0.9.2 Snapshot version of DLF4.

Thanks

**Comment by claeyzre** [comment]
After digging a bit more, the Flatten doesn't seem to work:

4 Convolutions yield me this kind of shape `(16, 64)`, after the MaxPooling with pool size 2the shapes are now `(8, 64)`. Then I use a Flatten to get a vector of dimension `512` for each of the four convolutions. Concatenating those dimensions with the 4 categorical variables yields the correct `2728`. Here the Flatten doesn't seem to work, so the Concatenate take the length of the results of the max pooling (64) and 64 * 4 + the dimensions of the 4 categorical variables yield `936`.

**Comment by ChangbingChen** [comment]
hi, which keras version do you use? 1.x.x or 2.x.x

**Comment by claeyzre** [comment]
2.x.x

**Comment by raver119** [comment]
Can you please post some code that reproduces this problem?

**Comment by maxpumperla** [comment]
@claeyzre the flatten failure has been fixed in #4935, but to make sure everything works well with your embeddings, can you please provide me with

a) your keras script generating the model or
b) a _smaller_ version of your h5 model. the current one is larger than 250mb and github prevents me from checking it into our test resources. shouldn't be difficult to scale the whole thing down a little.

**Comment by maxpumperla** [comment]
p.s. see this for tests related to your case:

https://github.com/deeplearning4j/deeplearning4j/blob/8d9bfa3129f528486e5a595f2a76d1a872e7af5b/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/weights/KerasWeightSettingTests.java#L37-L41

https://github.com/deeplearning4j/deeplearning4j/blob/8d9bfa3129f528486e5a595f2a76d1a872e7af5b/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/e2e/KerasModelEndToEndTest.java#L263-L266

**Comment by maxpumperla** [comment]
ok, with the addition of #4936 this should be closed. let me know if you still have issues.
